### ISSUE TITLE
Controls and materials clash

### ISSUE THREAD

**Issue by jasondhales** [idaholab/moose#15689]
## Bug Description
When I attempt to use Controls to turn Materials on/off that supply the same property, I get an error that two are supplying the same property.
[test.txt](https://github.com/idaholab/moose/files/5004195/test.txt)
That is true in a sense, but they aren't active at the same time.  (I can see that sorting this out would be difficult for the general case.)

## Steps to Reproduce
If you run the input file I attached, you'll see what I mean.  This example demonstrates the issue but is simple.  My use case is more complex.

## Impact
I haven't found a way to make this work that doesn't involve a hack of some sort.  If there is a way to swap out Materials on the fly that supply the same property, please let me know.

**Comment by andrsd** [comment]
Material `A` would supply property `prop1`, material `B` would supply `prop2`, material `C` supplies the wanted property and would switch between `prop1` and `prop2` based on your control evaluation.

**Comment by rwcarlsen** [comment]
The problem seems to be that the materials integrity check occurs before time starts ticking for the simulation.  And both of your material properties are active initially.  The control doesn't switch one off until after time starts ticking.  So during setup time, they are both on and the error gets thrown.  Just put `enable = false` for one of the materials in the input file.  And everything should work like you expect.

**Comment by andrsd** [comment]
@rwcarlsen I have not checked the code, so I am just asking. Are you saying that if I have 2 calls to `declareProperty` the second call will bind into the memory space the first material allocated? I remember this used to be an erroneous setup in MOOSE. If that's not the case anymore, I guess I missed the PR that changed it...

**Comment by rwcarlsen** [comment]
Yes - they bind into the same slot in the same MaterialData object.  Which is what we would want/expect here.  This isn't a problem since normally we don't allow two materials to provide the same property at the same time.  The controls system could cause this condition to be violated however if you later enable a duplicate property/material (e.g. a failed material object/property substitution).  Ideally, we would have some way of checking/maintaining this integrity during the simulation - not just checking once at setup time.
